### DISCUSSION TITLE
Modify a row after a delay

### DISCUSSION THREAD

**Discussion by rafikApp** [Budibase/budibase#7306] [Help] [answered]
Hello everyone.
Can we modify a row after a delay ?

**Comment by R2bEEaton** [comment, marked as answer]
Hi @rafikApp,

Instead of using a Save Row action when you click a button to save a row, for example, you could use a Trigger Automation instead. Then, create an automation triggered by an app action and include all the relevant data fields (eg. from a Form or an existing table row). You can follow this guide for that: https://docs.budibase.com/docs/app-action.

Now in the automation, you can set it up like so. The crucial part is the ["Delay" step](https://docs.budibase.com/docs/automation-actions#adding-logic). I chose `60000` milliseconds (60 seconds). This all has the effect of updating the name of a specified user 60 seconds after the app action is triggered. You can use other triggers for the automation, including CRON or when a row in a specified table updates. The sky is the limit!

If this solves your issue, please accept my answer using the check box above. Otherwise, let me know if you have further questions.

![image](https://user-images.githubusercontent.com/34921506/184908278-fe8b5adb-1f24-48af-a3dd-65b14387a282.png)
![image](https://user-images.githubusercontent.com/34921506/184908330-33d887e9-d33d-4a81-9440-6486f01ea842.png)
![image](https://user-images.githubusercontent.com/34921506/184908396-b7edd75e-7f0c-4966-b57b-70c0bc6b24e9.png)

**Reply by rafikApp** [reply]
No I want to save the row in the database, and after a 15 min I want to modify a column of this row

**Reply by R2bEEaton** [reply]
Yeah, so in that case you would create an Automation that is triggered by either a Row Creation or Update.

![image](https://user-images.githubusercontent.com/34921506/184949629-71fb098d-24f2-460f-8768-e113f58a067e.png)

Then you can set a 15 minute delay (900000 milliseconds) and then after that, have another action for "Update Row" which you use to update the desired column. You also need to pass the `_id` of the original row, which you can get easily with `{{ trigger.row._id }}` if you used a row trigger, for example.

**Reply by mmalessa80** [reply]
The proposed solution would modify a row after a delay. But you would need to keep in mind that this still needs to be triggered somehow. So the above solution just needs to be tweaked a little. Instead of App Action, the automation needs to be set to Row Created or Row Updated instead.

**Reply by rafikApp** [reply]
Thank you @R2bEEaton and @mmalessa80 
It works
